lib_collections/models.py

```diff
@@ -329,72 +330,23 @@ def get_context(self, request):
         except:
             pass
 
-        unit_title = ''
-        unit_url = ''
-        unit_email_label = ''
-        unit_email = ''
-        unit_phone_label = ''
-        unit_phone_number = ''
-        unit_fax_number = ''
-        unit_link_text = ''
-        unit_link_external = ''
-        unit_link_page = ''
-        unit_link_document = ''
-        if self.unit_contact:
-            try:
-                unit_title = self.unit.title
-            except:
-                pass
-
-            try:
-                unit_url = self.unit.public_web_page.url
-            except:
-                pass
-
-            try:
-                unit_email_label = self.unit.email_label
-            except:
-                pass
-            try:
-                unit_email = self.unit.email
-            except:
-                pass
-
-            try:
-                unit_phone_label = self.unit.unit_page_phone_number.first(
-                ).phone_label
-            except:
-                pass
-            try:
-                unit_phone_number = self.unit.unit_page_phone_number.first(
-                ).phone_number
-            except:
-                pass
-
-            try:
-                unit_fax_number = self.unit.fax_number
-            except:
-                pass
-
-            try:
-                unit_link_text = self.unit.link_text
-            except:
-                pass
-
-            try:
-                unit_link_external = self.unit.link_external
-            except:
-                pass
-
-            try:
-                unit_link_page = self.unit.link_page.url
-            except:
-                pass
-
-            try:
-                unit_link_document = self.unit.link_document.file.url
-            except:
-                pass
+        unit_title = lazy_dotchain(lambda: self.unit.title, '')
```

Comment:
@kzadrozny just for fun and FYI...

Python is an eager language. That means that arguments to functions are evaluated when they're passed to the function (not inside the function when they're used). This causes a problem because if any of these arguments such as `self.unit.link_document.file.url` don't exist, the code will blow up. That's why this code was previously in `try/except` blocks.

The `try/except` blocks cause the linter to complain because they're "bare". To fix them, they should be written as `except (AttributeError):`. I didn't want to do this because the code is repetitive. To simplify it, I wrote a function called `lazy_dotchain`. [This function is in my reusable library](https://github.com/bbusenius/Diablo-Python/blob/master/diablo_utils/diablo_utils.py) since I can use it on other projects. `lazy_dotchain` evaluates the existence of the arguments (e.g.`self.unit.link_document.file.url`) and returns the value if it finds it or a null value if it does not. In this case we set the null value to an empty string.

We get around python being an eager language by passing a `lambda`. Lambda functions are one of the few ways that python allows "laziness". In other words, since we pass the arguments to `lazy_dotchain` as `lambda` functions, they aren't evaluated until they're used inside `lazy_dotchain` instead of outside when they're passed. This allows me to set the `try/except` only once instead of multiple times like we did before.